### ISSUE TITLE
std::out_of_range

### ISSUE THREAD

**Issue by Starl1ght** [p4lang/behavioral-model#291]
Following code is produced by p4c-bm compiler, everything is OK. However, once I run it in bmv2, I get 

> terminate called after throwing an instance of 'std::out_of_range'
>   what():  _Map_base::at

I've attached produced JSON. Sorry for very big size, however, this problem suddenly appeared out of nowhere, and was not able to exactly pin-point it and produce minimal example
[proj.json.txt](https://github.com/p4lang/behavioral-model/files/782622/proj.json.txt)

**Comment by antoninbas** [comment]
This is because of the use of `<header_stack>[last]` in a set statement in the parser.
I was able to reproduce it with this simple P4 program:
```
header_type stack_t {
    fields { f1 : 8; f2 : 8; }
}

header stack_t stack[2];

header_type meta_t {
    fields { offset : 8; }
}

metadata meta_t meta;

parser start {
    extract(stack[next]);
    set_metadata(meta.offset, meta.offset + stack[last].f2);
    return ingress;
}

control ingress { }
```

**Comment by antoninbas** [comment]
This is strongly related to this know issue: https://github.com/p4lang/behavioral-model/issues/152

**Comment by Starl1ght** [comment]
Any possible workaround here? I do really need something like 

>  set_metadata(meta.offset, meta.offset + stack[last].f2);
for Ipv6 offsets parsing and other stuff :(

**Comment by antoninbas** [comment]
Fixed in #292
